Make merge gatekeeper more resilient

franklad opened the pull request:
Addressing Issue: #64 by adding a round tripper to perform retries.

rytswd commented on README.md:
```diff
@@ -16,7 +16,7 @@ At UPSIDER, we have a few internal repositories set up with a monorepo structure
 
 We are looking to add a few more features, such as extra signoff from non-coder, label based check, etc.
 
-<sup><sub>NOTE:  
```
If you take out the blank spaces here, it would change the line break layout slightly. While it seems to read just fine with this change, I'd like to check if this was an intentional update?

franklad replied:
my ide removed it

rytswd commented on README.md:
This is something very peculiar to our setup, but we are using Importer https://github.com/upsidr/importer for our documentation. This means we are only pulling in pieces of documentation from `docs` directory.

You can see some markdown comments such as 

```markdown
<!-- == imptr: inputs / begin from: ./docs/action-usage.md#[inputs] == -->
```

It means it's checking the file `./docs/action-usage.md` and a marker `inputs`, and when you run `importer update README.md`, it would replace the documentation with that given content. This is our own way of ensuring documentation does not get drifted between places, and that's the reason why the CI is failing. If you could make sure to update the `./docs` directory content as well as `README.md`, that would be appreciated! 🥰

franklad replied:
addressed

rytswd reviewed (commented):
I left a few comments about the markdown related CI failure 🙏

Also, the new logic looks OK, but it would be great if you could add some test cases to ensure we got the right implementation in place ☺️

rytswd commented:
Btw, I'm so sorry for the extremely delayed review 🙇
We will try to get our review cycles in place for future changes -- please do not hesitate to mention myself if you are not getting any response 🙏

franklad commented:
Hi @rytswd, thanks for your comments and sorry for my delayed response too!
I added some tests and updated the docs. would appreciate another look! cheers
